sdk/quoting/src/App.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react'
+import './App.css'
+import { ethers } from 'ethers'
+import { CurrentEnvironment } from './env'
+import Quoter from '@uniswap/v3-periphery/artifacts/contracts/lens/Quoter.sol/Quoter.json'
+import IUniswapV3PoolABI from '@uniswap/v3-core/artifacts/contracts/interfaces/IUniswapV3Pool.sol/IUniswapV3Pool.json'
+import { FeeAmount, computePoolAddress } from '@uniswap/v3-sdk'
+
+const rpcProvider = new ethers.providers.JsonRpcProvider(CurrentEnvironment.mainnetRpcUrl)
+const localRpcProvider = new ethers.providers.JsonRpcProvider(CurrentEnvironment.localRpcUrl)
+const POOL_FACTORY_CONTRACT_ADDRESS = '0x1F98431c8aD98523631AE4a59f267346ea31F984'
+const QUOTER_CONTRACT_ADDRESS = '0xb27308f9F90D607463bb33eA1BeBb41C27CE5AB6'
+
+const getPoolConstants = async () => {
+  const currentPoolAddress = computePoolAddress({
+    factoryAddress: POOL_FACTORY_CONTRACT_ADDRESS,
+    tokenA: CurrentEnvironment.tokenIn,
+    tokenB: CurrentEnvironment.tokenOut,
+    fee: FeeAmount.MEDIUM,
+  })
+
+  const poolContract = new ethers.Contract(
+    currentPoolAddress,
+    IUniswapV3PoolABI.abi,
+    CurrentEnvironment.isLocal ? localRpcProvider : rpcProvider
+  )
+
+  const [token0, token1, fee] = await Promise.all([poolContract.token0(), poolContract.token1(), poolContract.fee()])
+
+  return {
+    token0,
+    token1,
+    fee,
+  }
+}
+
+const quote = async (setOutputAmount: (outputAmount: number) => void) => {
+  const quoterContract = new ethers.Contract(
+    QUOTER_CONTRACT_ADDRESS,
+    Quoter.abi,
+    CurrentEnvironment.isLocal ? localRpcProvider : rpcProvider
+  )
+  const poolConstants = await getPoolConstants()
+
+  const quotedAmountOut = await quoterContract.callStatic.quoteExactInputSingle(
+    poolConstants.token0,
+    poolConstants.token1,
+    poolConstants.fee,
+    CurrentEnvironment.tokenInAmount * Math.pow(10, CurrentEnvironment.tokenIn.decimals),
```

Comment:
Note: I'm going to extract this out into a helper in the template, seems very common for people to need it!

Reply:
Plus one - I was sure you were going to add it regardless!